nilearn/decomposition/_base.py

```diff
@@ -0,0 +1,762 @@
+                    memory_level=0,
+                    memory=Memory(cachedir=None),):
+        if isinstance(imgs, nibabel.Nifti1Image):
+            imgs = [imgs]
+
+        if reduction_ratio != 'auto':
+            reduction_ratio = float(reduction_ratio)
+            if reduction_ratio is None or reduction_ratio >= 1:
+                reduction_ratio = 1
+        # Precomputing number of samples for preallocation
+        subject_n_samples = np.zeros(len(imgs), dtype='int')
+
+        for i, img in enumerate(imgs):
+            if reduction_ratio == 'auto':
+                subject_n_samples[i] = min(n_components,
+                                           check_niimg_4d(img).get_shape()[3])
+            else:
+                subject_n_samples[i] = int(ceil(check_niimg_4d(img).
+                                           get_shape()[3] * reduction_ratio))
+        subject_limits = np.zeros(subject_n_samples.shape[0]+1,
+                                  dtype='int')
+        subject_limits[1:] = np.cumsum(subject_n_samples)
+        n_voxels = np.sum(masker.mask_img_.get_data())
+        n_samples = subject_limits[-1]
+
+        # We initialize data in memory or on disk
+        if n_voxels * n_samples * 8 > 2e9:
+            temp_dir = _get_dataset_dir('temp')
+            filename = os.path.join(temp_dir, 'data')
+            data = np.memmap(filename, dtype='float64', order='F', mode='w+',
+                             shape=(n_samples, n_voxels))
+        else:
+            temp_dir = None
+            data = np.empty((n_samples, n_voxels), order='F', dtype='float64')
+
+        if confounds is None:
+            confounds = [None] * len(imgs)
+
+        masker_type = type(masker)
+        for i, (img, confound) in enumerate(zip(imgs, confounds)):
+            this_data = cache(masker_type.transform, memory,
+                              memory_level=memory_level,
+                              func_memory_level=3)\
+                (masker, img, confound)
+            if subject_n_samples[i] <= this_data.shape[0] // 4:
+                U, S, _ = cache(randomized_svd, memory,
+                                memory_level=memory_level,
+                                func_memory_level=3)(this_data.T,
+                                                     subject_n_samples[i],
+                                                     random_state=random_state)
+                U = U.T
+            else:
+                U, S, _ = cache(linalg.svd, memory,
+                                memory_level=memory_level,
+                                func_memory_level=3)(this_data.T,
+                                                     full_matrices=False)
+                U = U.T[:subject_n_samples[i]].copy()
+                S = S[:subject_n_samples[i]]
+            U = U * S[:, np.newaxis]
+            data[subject_limits[i]:subject_limits[i+1], :] = U
+        return data, temp_dir
+
+
+
+
+def session_pca(imgs, mask_img, parameters,
```

Comment:
I think that you meant to remove this function along with PCAMultiNiftiMasker. Could you do it so we don't review unused code?